massdash/loaders/OpenSwathXICParquetLoader.py

```diff
@@ -0,0 +1,93 @@
+"""
+massdash/loaders/OpenSwathXICParquetLoader
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+
+
+from typing import List, Dict, Union
+from os.path import basename
+from pandas.core.api import DataFrame as DataFrame
+import pandas as pd
+
+# Loaders
+from .GenericChromatogramLoader import GenericChromatogramLoader
+from .ResultsLoader import ResultsLoader
+from .access import OpenSwathXICParquetAccess
+# Structs
+from ..structs import TransitionGroup, TransitionGroupCollection
+# Utils
+from massdash.util import LOGGER
+
+class OpenSwathXICParquetLoader(GenericChromatogramLoader):
+
+    ''' 
+    Class for loading Chromatograms and peak features from SqMass files and OSW files
+    Inherits from GenericChromatogramLoader
+    '''
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs) 
+        self.dataAccess = [OpenSwathXICParquetAccess(f) for f in self.dataFiles]
+        
+    @ResultsLoader.cache_results
+    def loadTransitionGroupsDf(self, pep_id: str, charge: int) -> pd.DataFrame:
+        columns=['run_name', 'rt', 'intensity', 'annotation']
+        out = {}
+        for t in self.dataAccess:
+
+            df =  t.getChromatogramDfFromSequenceAndCharge(pep_id, charge)
+            # only add if there is data
+            if not df.empty:
+                out[t.runName] = df
+            else:
+                LOGGER.warning(f"Warning: no data found for peptide in transition file {t.filename}")
+
+        if out == {}:
+            return pd.DataFrame(columns=columns)
+        else:
+            return pd.concat(out).reset_index().drop('level_1', axis=1).rename(columns=dict(level_0='run'))
+
+    @ResultsLoader.cache_results
+    def loadTransitionGroups(self, pep_id: str, charge: int, runNames: Union[None, str, List[str]] =None) -> Dict[str, TransitionGroupCollection]:
+        '''
+        Loads the transition group for a given peptide ID and charge across all files
+        Args:
+            pep_id (str): Peptide ID
+            charge (int): Charge
+            runNames (None | str | List[str]): Name of the run to extract the transition group from. If None, all runs are extracted. If str, only the specified run is extracted. If List[str], only the specified runs are extracted.
+        Returns:
+            Dict[str, TransitionGroup]: Dictionary of TransitionGroups, with keys as sqMass filenames
+        '''
+
+        out = TransitionGroupCollection()
+        
+        def _assembleTransitionGroup(t):
+            chroms = t.getChromatogramsFromSequenceAndCharge(pep_id, charge)
+            precursorChroms = [i for i in chroms if 'precursor' in  i.label.lower()]
+            transitionChroms = [i for i in chroms if 'precursor' not in  i.label.lower()]
+            if len(precursorChroms) == 0 and len(transitionChroms) == 0: # do not create a transition group if there are no chromatograms
+                return None
```

Comment:
Is there an instance where we would want to return none, instead of throwing a message stating that no data was found for this peptide precursor?

Reply:
This is the behaviour that is used in the SqMassLoader so I did it this way for consistency. I believe I implemented it this way because  an empty Datatframe is returned for the loadTransitionGroupDf() method so this is somewhat consistent with that. 

 I could change the behaviour for both of them if you think that would be more appropriate.  

https://github.com/Roestlab/massdash/blob/a0e48a9270f7b6d2df9daa286dfa27982739ad66/massdash/loaders/SqMassLoader.py#L82-L83